mdpow/fep.py

```diff
@@ -625,46 +615,31 @@ def _setup(self, component, lmbda, foreign_lambdas=None, **kwargs):
-            # desired method, defaults to TI otherwise
-            if foreign_lambdas[0] is None:
-                feplambdas = foreign_lambdas[1]
-            elif foreign_lambdas[1] is None:
-                feplambdas = foreign_lambdas[0]
-            else:
-                feplambdas = "{0} {1}".format(foreign_lambdas[0], foreign_lambdas[1])
-            kwargs.update(dirname=wdir, struct=self.struct, top=self.top,
-                          mdp=self.mdp,
-                          ndx=self.ndx,
-                          mainselection=None,
-                          runtime=self.runtime,
-                          ref_t=self.Temperature,    # TODO: maybe not working yet, check _setup()
-                          gen_temp=self.Temperature, # needed until gromacs.setup() is smarter
-                          qname=self.tasklabel(component,lmbda),
-                          free_energy='yes',
-                          couple_moltype=self.molecule,
-                          init_lambda=lmbda,
-                          fep_lambdas=feplambdas,
-                          )
-
-        # for TI
+        
+        if kwargs.pop('edr', True):
+            logger.info('Setting dhdl file to edr format')
+            kwargs.setdefault('separate-dhdl-file', 'no')
         else:
-            kwargs.update(dirname=wdir, struct=self.struct, top=self.top,
-                          mdp=self.mdp,
-                          ndx=self.ndx,
-                          mainselection=None,
-                          runtime=self.runtime,
-                          ref_t=self.Temperature,    # TODO: maybe not working yet, check _setup()
-                          gen_temp=self.Temperature, # needed until gromacs.setup() is smarter
-                          qname=self.tasklabel(component,lmbda),
-                          free_energy='yes',
-                          couple_moltype=self.molecule,
-                          init_lambda=lmbda,
-                          delta_lambda=0,
-                          )
+            logger.info('Setting dhdl file to xvg format')
+            kwargs.setdefault('separate-dhdl-file', 'yes')
+
+        lambda_index = numpy.where(foreign_lambdas == lmbda)[0][0]
+
+        kwargs.update(dirname=wdir, struct=self.struct, top=self.top,
+                      mdp=self.mdp,
+                      ndx=self.ndx,
+                      mainselection=None,
+                      runtime=self.runtime,
+                      ref_t=self.Temperature,    # TODO: maybe not working yet, check _setup()
+                      gen_temp=self.Temperature, # needed until gromacs.setup() is smarter
+                      qname=self.tasklabel(component,lmbda),
+                      free_energy='yes',
+                      couple_moltype=self.molecule,
+                      init_lambda_state=lambda_index,
+                      fep_lambdas=foreign_lambdas,
+                      calc_lambda_neighbors=-1,
```

Comment:
Does `calc_lambda_neighbors=-1` work for Gromacs 4.6.x?

If not, you need code for Gromacs 5.x/2016 and Gromacs 4.6.x